### DISCUSSION TITLE
Async parsers

### DISCUSSION THREAD

**Discussion by nicolo-ribaudo** [eslint/eslint#14295] [Q&A] [answered]
Hi!
I'm slowly converting Babel to ESM, hopefully releasing an ESM-only version when Node.js 10 stops being maintained.

However, I stumbled upon a problem when making `@babel/eslint-parser` work with the ESM version of Babel: Babel's config loading will be mostly asynchronous, and I don't know how to make it work with an ESLint parser. I found https://github.com/eslint/rfcs/tree/master/designs/2019-move-to-async-api which makes me think that async parsers aren't completely impossible, but I didn't find anything about them.

It's something already supported, or is it already discussed, or should I open a new issue?

Thanks!

**Comment by nzakas** [comment, marked as answer]
We don’t have async parsers on the roadmap, and I’m not sure there’s a great use case for that. I’d suggest talking with the Babel folks about this problem as it doesn’t make sense to publish a parser for use with ESLint that doesn’t work with ESLint.

**Reply by nzakas** [reply]
Oops, skimmed over and turn realized you were the one making the change. Sorry about that.

My reply is still the same, though: we don’t have async parsers on the roadmap and parsing is so deep inside of ESLint that it would be a large effort for just one parser that needs it.

**Reply by nicolo-ribaudo** [reply]
Thanks for the reply!

We are researching if it would be possible to release Babel as ESM starting from the next major version (which means that loading the parser would be asynchronous). We'll try to figure out a workaround, in the worst case we can keep the pieces needed by `@babel/eslint-parser` as CJS.
